src/devices_models/devices/hydro_generation.jl

```diff
@@ -140,109 +140,72 @@ function commitment_constraints!(
+        return DeviceRangeConstraintInputs(;
+            range_constraint_inputs = [RangeConstraintInputs(;
+                constraint_name = ACTIVE_RANGE,
+                variable_name = ACTIVE_POWER,
+                limits_func = x -> (min = 0.0, max = PSY.get_activepower(x)),
+                constraint_func = device_range,
+            )],
         )
-        return
-    end
-
-    forecast_label = "get_rating"
-    constraint_infos = Vector{DeviceTimeSeriesConstraintInfo}(undef, length(devices))
-    for (ix, d) in enumerate(devices)
-        ts_vector = get_time_series(psi_container, d, forecast_label)
-        constraint_info =
-            DeviceTimeSeriesConstraintInfo(d, x -> PSY.get_rating(x), ts_vector)
-        add_device_services!(constraint_info.range, d, model)
-        constraint_infos[ix] = constraint_info
     end
 
-    if parameters
-        device_timeseries_param_ub(
-            psi_container,
-            constraint_infos,
-            constraint_name(ACTIVE, H),
-            UpdateRef{H}(ACTIVE_POWER, forecast_label),
-            variable_name(ACTIVE_POWER, H),
-        )
-    else
-        device_timeseries_ub(
-            psi_container,
-            constraint_infos,
-            constraint_name(ACTIVE, H),
-            variable_name(ACTIVE_POWER, H),
-        )
-    end
-    return
+    return DeviceRangeConstraintInputs(;
+        timeseries_range_constraint_inputs = [TimeSeriesConstraintInputs(
+            constraint_name = ACTIVE,
+            variable_name = ACTIVE_POWER,
+            parameter_name = use_parameters ? ACTIVE_POWER : nothing,
+            forecast_label = "get_rating",
+            multiplier_func = x -> PSY.get_rating(x),
+            constraint_func = use_parameters ? device_timeseries_param_ub :
+                              device_timeseries_ub,
+        )],
+    )
 end
 
-function activepower_constraints!(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{H},
-    model::DeviceModel{H, <:AbstractHydroReservoirFormulation},
-    system_formulation::Type{<:PM.AbstractPowerModel},
-    feedforward::Union{Nothing, AbstractAffectFeedForward},
-) where {H <: PSY.HydroGen}
-    constraint_infos = Vector{DeviceRangeConstraintInfo}(undef, length(devices))
-    for (ix, d) in enumerate(devices)
-        name = PSY.get_name(d)
-        limits = PSY.get_activepowerlimits(d)
-        constraint_info = DeviceRangeConstraintInfo(name, limits)
-        add_device_services!(constraint_info, d, model)
-        constraint_infos[ix] = constraint_info
-    end
-    device_range(
-        psi_container,
-        constraint_infos,
-        constraint_name(ACTIVE_RANGE, H),
-        variable_name(ACTIVE_POWER, H),
+function make_active_power_constraints_inputs(
+    ::Type{<:PSY.HydroGen},
+    ::Type{<:AbstractHydroReservoirFormulation},
+    ::Type{<:PM.AbstractPowerModel},
+    _::Union{Nothing, AbstractAffectFeedForward},
+    __::Bool,
+    ___::Bool,
```

Comment:
is this whitespace? I don't get what GitHub is showing here

Reply:
Are you referring to the underscores?  That is a way to inform the user that a variable is unused.  In Python pylint will not complain about them being unused.  If you put actual names, it will.  Others label these as `unused`.  Are you OK with this?

Reply:
I was just curious since its the first time I see it and we regularly don't use this.